force-app/main/default/classes/DMLOperations.cls

```diff
@@ -181,13 +230,24 @@ public with sharing class DMLOperations {
 	 */
 	public static void upsertAccountsWithContacts(List<Contact> contactsToUpsert) {
 		// Loop through each Contact
-
+        for(Contact con:contactsToUpsert){
```

Comment:
Try following these steps to move your DML and queries outside of the loops. Its a lot more challenging in this particular method, which is why the instructions say you can utilize the previous method. 

1. Collect unique last names from the list of Contacts to upsert
2. Query existing Accounts with names matching the last names
3. Create new Accounts for last names that do not have a matching Account
4. Associate Contacts with their corresponding Accounts
5. Upsert the Contacts